### PR TITLE
Print transfer plugins versions on transfer-config

### PR DESCRIPTION
**Pull request by asafgabai**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Printing the transfer plugins' versions on transfer-config, and log improvements.

### CODE REVIEW

File: artifactory/commands/transferconfig/transferconfig.go [hunk 1]
```diff
@@ -62,6 +64,7 @@ func (tcc *TransferConfigCommand) SetIncludedRepositories(includedRepositories [
 }
 
 func (tcc *TransferConfigCommand) Run() (err error) {
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold("========== Phase 1/4 - preparations ==========")))
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/transferconfig.go:67]: preparations --> Preparations

File: artifactory/commands/transferconfig/transferconfig.go [hunk 2]
```diff
@@ -77,12 +80,26 @@ func (tcc *TransferConfigCommand) Run() (err error) {
...
+	}
+	log.Info("data-transfer plugin version: " + dataTransferPluginVersion)
+
 	// Make sure that the source and target Artifactory servers are different and that the target Artifactory is empty
 	if err = tcc.validateArtifactoryServers(targetServiceManager, sourceArtifactoryVersion); err != nil {
 		return
 	}
 
 	// Run export on the source Artifactory
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold("========== Phase 2/4 - export configuration from the source Artifactory ==========")))
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/transferconfig.go:102]: export --> Export

File: artifactory/commands/transferconfig/transferconfig.go [hunk 3]
```diff
@@ -94,6 +111,8 @@ func (tcc *TransferConfigCommand) Run() (err error) {
 		return
 	}
 
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold("========== Phase 3/4 - download and modify configuration ==========")))
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/transferconfig.go:114]: download --> Download

File: artifactory/commands/transferconfig/transferconfig.go [hunk 4]
```diff
@@ -113,6 +132,7 @@ func (tcc *TransferConfigCommand) Run() (err error) {
 	}
 
 	// Import the archive to the target Artifactory
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold("========== Phase 4/4 - import configuration to the target Artifactory ==========")))
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/transferconfig.go:135]: import --> Import

File: artifactory/commands/transferconfig/transferconfig.go [hunk 5]
```diff
@@ -363,3 +386,31 @@ func (tcc *TransferConfigCommand) createImportPollingAction(targetServicesManage
...
+		if err != nil {
+			return "", errorutils.CheckError(err)
+		}
+		return verRes.Version, nil
+	}
+
+	messageFormat := fmt.Sprintf("Response from Artifactory: %s.\n%s\n", versionResp.Status, versionBody)
+	if versionResp.StatusCode == http.StatusNotFound {
+		// Probably the plugin is not installed
+		return "", errorutils.CheckErrorf("%sIt looks like the %s plugin is not installed on the server.", messageFormat, pluginName) /////////////TODO: change the message!!!!!
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/transferconfig.go:411]: Let's remove the above comment.

File: artifactory/commands/transferconfig/transferconfig_test.go [hunk 6]
```diff
@@ -99,15 +99,20 @@ func TestSanityVerifications(t *testing.T) {
 	testServer, serverDetails, serviceManager := createMockServer(t, func(w http.ResponseWriter, r *http.Request) {
 		if r.RequestURI == "/api/plugins/execute/checkPermissions" {
 			w.WriteHeader(http.StatusOK)
+		} else if r.RequestURI == "/api/plugins/execute/configImportVersion" {
```
Comment (eyalbe4) [inline artifactory/commands/transferconfig/transferconfig_test.go:102]: Do we have testing code that calls the above `configImportVersion` mock?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: artifactory/commands/transferconfig/transferconfig_test.go (hunk 6 above)
Comment (asafgabai) [reply artifactory/commands/transferconfig/transferconfig_test.go:102]: Yes. The function validateArtifactoryServers looks for the plugin's version and prints it. It must return 200.

**Pull request merged**